rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2500,6 +2500,86 @@ void getProposedFederatorPublicKeyOfType_whenStorageProviderReturnsProposedFeder
         assertArrayEquals(expectedPublicKey, actualPublicKey.get());
     }
 
+    @Nested
+    @Tag("Federation is in or past migration age")
+    class FederationMigrationAge {
+        private static final long NEW_FEDERATION_CREATION_BLOCK = 100L;
+        private static final long FEDERATION_ACTIVATION_AGE = federationMainnetConstants.getFederationActivationAge(allActivations);
+        private static final long FUNDS_MIGRATION_AGE_SINCE_ACTIVATION_BEGIN = federationMainnetConstants.getFundsMigrationAgeSinceActivationBegin();
+        private static final long ONE_BLOCK_BEFORE_MIGRATION_BEGINS = FEDERATION_ACTIVATION_AGE + FUNDS_MIGRATION_AGE_SINCE_ACTIVATION_BEGIN;
+        private static final long IN_MIGRATION_AGE_EXECUTION_BLOCK_NUMBER = NEW_FEDERATION_CREATION_BLOCK + ONE_BLOCK_BEFORE_MIGRATION_BEGINS + 1;
+        private static final long FUNDS_MIGRATION_AGE_SINCE_ACTIVATION_ENDS = federationMainnetConstants.getFundsMigrationAgeSinceActivationEnd(allActivations);
+        private static final long ONE_BLOCK_AFTER_MIGRATION_ENDS_BLOCK_NUMBER = NEW_FEDERATION_CREATION_BLOCK + ONE_BLOCK_BEFORE_MIGRATION_BEGINS + FUNDS_MIGRATION_AGE_SINCE_ACTIVATION_ENDS;
+        private static final Federation newFederation = P2shP2wshErpFederationBuilder.builder().withCreationBlockNumber(NEW_FEDERATION_CREATION_BLOCK).build();
+
+        @BeforeEach
+        void setUp() {
+            storageProvider = new FederationStorageProviderImpl(storageAccessor);
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .build();
```

Comment:
```suggestion
         private Block block;
         
        @BeforeEach
        void setUp() {
            storageProvider = new FederationStorageProviderImpl(storageAccessor);
            block = mock(Block.class);
            federationSupport = federationSupportBuilder
                .withFederationConstants(federationMainnetConstants)
                .withFederationStorageProvider(storageProvider)
                .withRskExecutionBlock(block)
                .build();
```

this way you can do the specific mocking in each test without rebuilding the fed support